packages/neo4j/tests/index.test.ts

```diff
@@ -42,7 +42,7 @@ runBasicTests({
       const session = result?.records[0]?.get("s")?.properties
       const userId = result?.records[0]?.get("userId")
 
-      if (!session || !userId) return null
+      if (!session || session.userId || !userId) return null
```

Comment:
Why is this necessary?

Reply:
I think this and the `|| account.userId` are needed to catch the case where the userId property has been accidentally set by the adapter. For example, if the test was like this during your refactor it might have helped you catch the change that I made in this PR re: `async createSession(data) {`.

There is nothing to say that another developer might come along and get hit by the same gotcha. I mean, adding all the data to the Session node does look like a logical way to clean up the code. This clause just gaurds against it.

Reply:
It sounds unnecessary to guard against that at runtime for every user. It did not produce an error in the tests, so technically this is not even a problem IMO.

If you think the tests should catch this, maybe we should add some extra tests at the bottom of the test file instead.

Reply:
This is the tests file so it doesn't run at runtime. I did think about additonal tests but it's a bunch more LOC for something pretty simple.

Reply:
My bad... :sweat_smile: